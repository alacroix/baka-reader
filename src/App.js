import { StackNavigator } from 'react-navigation';

import * as Pages from 'bakareader/src/pages';

export default StackNavigator({
  appLoader: {
    screen: Pages.AppLoader,
  },
  home: {
    screen: Pages.Home,
  },
  infos: {
    screen: Pages.Infos,
  },
}, {
  initialRouteName: 'appLoader',
});
