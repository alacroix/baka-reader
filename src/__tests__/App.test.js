import 'react-native';
import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';

import App from '../App';

const renderer = new ReactShallowRenderer();

jest.mock('../services/FileSystem');

const filterKeys = (state) => {
  if (state.routes) {
    return {
      ...state,
      routes: state.routes.map((route) => {
        const { key, ...others } = route;
        return filterKeys(others);
      }),
    };
  }
  return state;
};

describe('App', () => {
  it('should render itself', () => {
    renderer.render(<App />);
    const tree = renderer.getRenderOutput();
    tree.props.navigation.state = filterKeys(tree.props.navigation.state);
    expect(tree).toMatchSnapshot();
  });
});
