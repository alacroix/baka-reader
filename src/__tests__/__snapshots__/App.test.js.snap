// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render itself 1`] = `
<Navigator
  navigation={
    Object {
      "dispatch": [Function],
      "goBack": [Function],
      "navigate": [Function],
      "setParams": [Function],
      "state": Object {
        "index": 0,
        "routes": Array [
          Object {
            "routeName": "appLoader",
          },
        ],
      },
    }
  }
/>
`;
